src/main/java/Problem3/Problem3.java

```diff
@@ -8,9 +8,20 @@
 
 public class Problem3 {
     public static void printListInReverse(ListNode list, PrintStream print) {
-        // homework
-        // !!! must use your ArrayStack or LinkedListStack in problem 1
-        // print with the following format. System.out.println() will not pass test.
-        print.println("hello world");   // place holder
+
+        LinkedListStack<ListNode> data = new LinkedListStack<>();
+
+        //Puts the list into the data LinkedListStack
+        ListNode pl = list.next;
+        while (pl != null)
+        {
+            data.push(pl);
+            pl = pl.next;
+        }
+
+        while (data.size() > 0) {
+            print.println(data.peek().val);
+            data.pop();
```

Comment:
pop also return values same as peek so on line 23 you could also use pop() and get rid of line 24.